civihr_hrjobcontract_entities/civihr_hrjobcontract_entities.module

```diff
@@ -251,6 +251,43 @@ function civihr_hrjobcontract_entities_init() {
                     WHERE c.deleted = 0
                     ORDER BY t.job_contract_id");
 
+
+        db_query('DROP VIEW IF EXISTS hrjc_contact_details');
+        db_query("CREATE OR REPLACE VIEW hrjc_contact_details AS
+          SELECT c.id AS contact_id,
+            GROUP_CONCAT(DISTINCT CASE WHEN phone_location.name = 'Work' THEN c_phone.phone ELSE '' END
+              ORDER BY c_phone.phone DESC SEPARATOR ' ') AS work_phone,
+            GROUP_CONCAT(DISTINCT CASE WHEN phone_location.name = 'Work' THEN c_phone.phone_ext ELSE '' END 
+              ORDER BY c_phone.phone_ext DESC SEPARATOR ' ') AS work_phone_ext,
+            GROUP_CONCAT(DISTINCT CASE WHEN email_location.name = 'Work' THEN c_email.email ELSE '' END 
+              ORDER BY c_email.email DESC SEPARATOR ' ') AS work_email,
+
+            GROUP_CONCAT(DISTINCT CASE WHEN phone_location.name = 'Home' THEN c_phone.phone ELSE '' END
```

Comment:
I could use an IN statement, but if I did, I'd end up having different records for work phone and home phone. Also for work e-mail and home e-mail. Worse case scenario, I'd have 4 different records to contend with.

On the other hand, to use the information in the views I'd have to include the relationship to the entity and then filter by what I need.  But if I filter on the view, all other records will not appear.  For example, if I filter for records with work e-mail, only records with work e-mail would appear, because views makes me filter adding clauses to the WHERE clause. What I need is that records with no work e-mail still appear, but with empty information. This could be done if I could add filters to the JOIN clause, but I found no easy way of doing this.  So I just built the query so that I get one record per contact with all the information I need.